crates/vm-superio/src/serial.rs

```diff
@@ -914,4 +1013,110 @@ mod tests {
         // THR empty interrupt was raised nevertheless.
         assert_eq!(iir & IIR_THR_EMPTY_BIT, IIR_THR_EMPTY_BIT);
     }
+
+    #[test]
+    fn test_serial_state_constructor() {
+        let state = SerialState {
+            baud_divisor_low: DEFAULT_BAUD_DIVISOR_LOW,
+            baud_divisor_high: DEFAULT_BAUD_DIVISOR_HIGH,
+            interrupt_enable: DEFAULT_INTERRUPT_ENABLE,
+            interrupt_identification: DEFAULT_INTERRUPT_IDENTIFICATION,
+            line_control: DEFAULT_LINE_CONTROL,
+            line_status: DEFAULT_LINE_STATUS,
+            modem_control: DEFAULT_MODEM_CONTROL,
+            modem_status: DEFAULT_MODEM_STATUS,
+            scratch: DEFAULT_SCRATCH,
+            in_buffer: Vec::new(),
+        };
+
+        assert_eq!(state.baud_divisor_low, DEFAULT_BAUD_DIVISOR_LOW);
+        assert_eq!(state.baud_divisor_high, DEFAULT_BAUD_DIVISOR_HIGH);
+        assert_eq!(state.interrupt_enable, DEFAULT_INTERRUPT_ENABLE);
+        assert_eq!(
+            state.interrupt_identification,
+            DEFAULT_INTERRUPT_IDENTIFICATION
+        );
+        assert_eq!(state.line_control, DEFAULT_LINE_CONTROL);
+        assert_eq!(state.line_status, DEFAULT_LINE_STATUS);
+        assert_eq!(state.modem_control, DEFAULT_MODEM_CONTROL);
+        assert_eq!(state.modem_status, DEFAULT_MODEM_STATUS);
+        assert_eq!(state.scratch, DEFAULT_SCRATCH);
+        assert!(state.in_buffer.is_empty());
+    }
+
+    #[test]
+    fn test_serial_state_default() {
+        let state = SerialState::default();
```

Comment:
I think it would be more useful if we instead create the serial from the default state (using `with_events`), then call `save_state`, and check that the state is the `default_state` using `assert_eq` (because `SerialState` derives `PartialEq`). I don't think we need to check the actual values because this is a duplication of the code.